lib/src/util/cast_ui_factory.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:over_react/component_base.dart';
+
+/// Casts [value] to `UiFactory<T>`, which can be inferred from the surrounding context,
+/// allowing you to perform casts of dynamic values without having to repeat the type.
```

Comment:
I think the information around the formatting/ignore comments might be worth moving up here so that we're talking about the advantages all in one place: 
```suggestion
/// allowing you to perform casts of dynamic values without having to repeat the type
/// or worry about how formatting affects ignore comment placement.
```